### PR TITLE
Parse BOM from GCS to get service config

### PR DESCRIPTION
**Pull request by lwander**
@jtk54 @duftler @ewiseblatt PTAL

### CODE REVIEW

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/spinnaker/v1/BillOfMaterials.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.Optional;
+
+@Data
+public class BillOfMaterials {
+  String version;
+  SpinnakerComponents services;
+
+  @Data
+  static class SpinnakerComponents {
+    String clouddriver;
```
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/spinnaker/v1/BillOfMaterials.java]: Described here: https://github.com/spinnaker/halyard/blob/master/docs/design.md#11-versioning-spinnaker

**Comment by lwander** [comment]
@ewiseblatt : https://github.com/spinnaker/halyard/blob/master/docs/design.md#1-configuring-spinnaker

**Comment by lwander** [comment]
I would treat this as a proof-of-concept, putting the pieces together we need from the release process with halyard to generate config.

Essentially, it checks a known endpoint for a bill of materials matching the version of Spinnaker to deploy. Once it has that bill of materials, it looks up whatever subcomponent's version it needs, and uses that version to get its config.

File: halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/ConfigController.java [hunk 2]
```diff
@@ -32,6 +34,9 @@
   @Autowired
   ConfigService configService;
 
+  @Autowired
+  ComponentProfile componentProfile;
```
Comment (jtk54) [inline halyard-web/src/main/java/com/netflix/spinnaker/halyard/controllers/v1/ConfigController.java]: Remove or use : )

**Comment by lwander** [comment]
I'm going to merge this for now - it's in line with out we set out to treat the BOM, but realistically only a small bit of code that's easy to replace if we come up with a different versioning policy.

**Pull request merged**